### PR TITLE
Header refactoring - prepare to kill the extent -> arena cycle

### PR DESCRIPTION
**Pull request by davidtgoldblatt**
We make rtree, the extent helpers, and mutex pool unified, and pull them out of the catchall.

This also pulls the size-computation helpers out of the jemalloc module, since they're relatively self contained and everything depends on them.

### CODE REVIEW

File: include/jemalloc/internal/jemalloc_internal_includes.h [hunk 1]
```diff
@@ -12,6 +12,8 @@
  * following #defines:
  *   JEMALLOC_H_TYPES   : Preprocessor-defined constants and psuedo-opaque data
  *                        types.
+
```
Comment (jasone) [inline include/jemalloc/internal/jemalloc_internal_includes.h]: Extraneous whitespace?

File: include/jemalloc/internal/size_classes.sh [hunk 2]
```diff
@@ -334,6 +334,7 @@ for lg_z in ${lg_zarr} ; do
         echo "#define LOOKUP_MAXCLASS		${lookup_maxclass}"
         echo "#define SMALL_MAXCLASS		${small_maxclass}"
         echo "#define LG_LARGE_MINCLASS	${lg_large_minclass}"
+	echo "#define LARGE_MINCLASS		(ZU(1) << LG_LARGE_MINCLASS)"
```
Comment (jasone) [inline include/jemalloc/internal/size_classes.sh]: Replace leading tab with spaces for consistency.

File: include/jemalloc/internal/jemalloc_internal_includes.h (hunk 1 above)
Comment (davidtgoldblatt) [inline include/jemalloc/internal/jemalloc_internal_includes.h]: Huh, yeah. fixed.

File: include/jemalloc/internal/size_classes.sh (hunk 2 above)
Comment (davidtgoldblatt) [inline include/jemalloc/internal/size_classes.sh]: Done.

**Comment by davidtgoldblatt** [comment]
Only whitespace changes from the last version, which passed all tests. Merging.

**Pull request merged**